src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/database/jpa/mapper/induction/InductionEntityMapper.kt

```diff
@@ -88,59 +88,61 @@ abstract class InductionEntityMapper {
 
   fun updateFutureWorkInterests(entity: InductionEntity, dto: UpdateInductionDto): FutureWorkInterestsEntity? {
     return if (entity.futureWorkInterests == null) {
-      futureWorkInterestsEntityMapper.fromUpdateDtoToEntity(dto.futureWorkInterests)
+      // if the Induction previously didn't have FutureWorkInterests (e.g. because the prisoner didn't want to work), then create a new FutureWorkInterests
+      futureWorkInterestsEntityMapper.fromUpdateDtoToNewEntity(dto.futureWorkInterests)
     } else {
-      futureWorkInterestsEntityMapper.updateEntityFromDto(entity.futureWorkInterests, dto.futureWorkInterests)
+      // else update the existing FutureWorkInterests
+      futureWorkInterestsEntityMapper.updateExistingEntityFromDto(entity.futureWorkInterests!!, dto.futureWorkInterests)
         .let { entity.futureWorkInterests }
```

Comment:
As you can see, I've done 3 things:

1. Renamed the methods to try and make their purpose more obvious.
2. Added the inline comments to explain the need for the if statement.
3. Made the entity a mandatory argument within the `updateExistingEntityFromDto()` methods.